src/autoscaler/db/sqldb/policy_sqldb.go

```diff
@@ -145,3 +146,41 @@ func (pdb *PolicySQLDB) DeletePolicy(appId string) error {
 	}
 	return err
 }
+
+func (pdb *PolicySQLDB) GetCustomMetricsCreds(appId string) (string, string, error) {
+	var password string
+	var username string
+	query := "SELECT username,password from credentials WHERE id = $1"
+	err := pdb.sqldb.QueryRow(query, appId).Scan(&username, &password)
+
+	if err != nil {
+		pdb.logger.Error("get-custom-metrics-creds-from-credentials-table", err, lager.Data{"query": query})
+		return "", "", err
+	}
+	return username, password, nil
+}
+
+func (pdb *PolicySQLDB) ValidateCustomMetricTypes(appGUID string, metricsConsumer *models.MetricsConsumer) (bool, error) {
+	scalingPolicy, err := pdb.GetAppPolicy(appGUID)
+	if err != nil {
+		pdb.logger.Error("errorr-getting-policy", err, lager.Data{"appId": appGUID})
+		return false, errors.New("not able to get policy details")
+	}
+	if err == nil && scalingPolicy == nil {
+		pdb.logger.Debug("no-policy-found", lager.Data{"appId": appGUID})
+		return false, errors.New("no policy found")
+	}
+	allowedMetricTypeSet := make(map[string]struct{}, len(scalingPolicy.ScalingRules))
```

Comment:
If my memory is right, we discussed to save "memoryused" "memoryutil" "responsetime" and "throughput" as the reserved keywords. it is necessary  to reject these metrics type as well

Reply:
Done